src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -218,17 +219,29 @@ private Control FindFirstAncestorControl()
 				? current.animation
 				: current.transition;
 			var nextAnimation = target.transition ?? target.animation;
-			if (runningAnimation != null)
+
+			// We need to dispatch here. The scenario is calling GoToState successively with two different states.
+			// In this case, we dispatch/schedule a begin for the first animation, then the second.
+			// The second GoToState call will come here and try to stop the running animation of the first call.
+			// But the first call wasn't yet run by the dispatcher, causing race condition and we go into unrecoverable bad state.
```

Comment:
I don't get that: Usually adding dispatch is only getting worst in term of concurrency issue and actually dispatching here would cause a concurrency issue IMO has we are not cancelling the previous dispatch if we invoke goToState twice.